lib/pxcontext.js

```diff
@@ -56,9 +56,12 @@ class PxContext {
                 }
             });
         }
-        if (this.uri.includes('graphql')) {
+        if (pxUtil.isGraphql(req, config)) {
             config.logger.debug('Graphql route detected');
-            this.graphqlData = pxUtil.getGraphqlData(req);
+            this.graphqlData = Array.isArray(req.body) ?
```

Comment:
You can make it cleaner by extracting this to a method called getGraphqlData which called to a method called extractGraphqlData.